cdap-common/src/main/java/co/cask/cdap/common/logging/LogSamplers.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.common.logging;
+
+import co.cask.cdap.common.utils.TimeProvider;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Supplier;
+import com.google.common.collect.Iterables;
+import com.google.common.util.concurrent.AtomicDouble;
+import org.slf4j.spi.LocationAwareLogger;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class to provide common implementations of {@link LogSampler}.
+ */
+public final class LogSamplers {
+
+  /**
+   * Returns a {@link LogSampler} that accepts once on every N calls.
+   */
+  public static LogSampler limit(final int count) {
```

Comment:
not a big deal, but 'limit' is a bit of an odd name to me. Without looking at the javadoc or implementation, I would think it logs at most N messages. How about something like 'modulo', or 'onceEvery'